sqlglot/parser.py

```diff
@@ -1346,6 +1346,32 @@ def _parse_command(self) -> exp.Command:
             exp.Command, this=self._prev.text.upper(), expression=self._parse_string()
         )
 
+    def _try_parse(
+        self, parse_method: t.Callable, retreat: bool = False
+    ) -> t.Optional[exp.Expression]:
+        """
+        Attemps to backtrack if a parse function that contains a try/catch internally raises an error. This behavior can
+        be different depending on the uset-set ErrorLevel, so _try_parse aims to solve this by setting & resetting
+        the parser state accordingly
+        """
+        index = self._index
+        errors = self.errors.copy()
+        error_level = self.error_level
+
+        self.error_level = ErrorLevel.IMMEDIATE
+        this = None
```

Comment:
i don't think this is necessary

Reply:
If the `error_level` isn't `IMMEDIATE`, the `try/catch` statements that are contained in `parse_method` might not fire at all (or too late, as in `RAISE`), right?

Reply:
We explicitly set the error level here to `IMMEDIATE` though, right?